app/src/editor/canvas/index.jsx

```diff
@@ -172,15 +172,52 @@ const CanvasEditComponent = class CanvasEdit extends Component {
     }
 
     loadNewDefinition = async (hash) => {
-        const { replace } = this.props
+        const { replace, undo } = this.props
         const module = CanvasState.getModule(this.props.canvas, hash)
-        const newModule = await sharedServices.getModule(module)
+        try {
+            const newModule = await sharedServices.getModule(module)
 
-        if (!this.unmounted) {
-            replace(() => CanvasState.updateModule(this.props.canvas, hash, () => newModule))
+            if (!this.unmounted) {
+                replace(() => CanvasState.updateModule(this.props.canvas, hash, () => newModule))
+            }
+        } catch (error) {
+            console.error(error.message)
+            // undo value change
+            undo()
```

Comment:
`undo()` is probably not what we want, since it would leave the bad state in the available to be redone… plus I'm not sure there's currently any guarantee of what exactly is being undone.

Reply:
Yep, it's probably not right. The issue here is that if the user is allowed to select a canvas without any inputs, the value is saved which will cause the whole canvas saving to fail. That's why we'd need something to reset the value back to a working canvas. Of course, best would be that the canvases wouldn't come from the backend in the first place...